### PR TITLE
Property for closing a link

### PR DESCRIPTION
**Pull request by MrRamka**
Implementation https://github.com/microsoft/charticulator/issues/844

image
image

### CODE REVIEW

File: src/core/prototypes/links/utils.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  if (!parent) {
+    return false;
+  }
+
+  if (!parent.object.elements) {
+    return false;
+  }
+
+  if (linkProperties.linkThrough) {
+    const plotSegment = parent.object.elements.find(
```
Comment (jrthompson33) [inline src/core/prototypes/links/utils.ts:26]: Is it possible for the parent chart element to have multiple elements that satisfy this condition? Here `find` will return the first element that satisfies the linkThrough condition. For example, multiple Polar plot lines on the same chart?

File: src/core/prototypes/links/utils.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// Licensed under the MIT license.
+
+import { ChartClass } from "../charts";
+import { LinksProperties } from "./index";
+import { PolarPlotSegment } from "../plot_segments";
+
+export function shouldShowCloseLink(
+  parent: ChartClass,
+  linkProperties: LinksProperties,
+  close?: boolean
```
Comment (jrthompson33) [inline src/core/prototypes/links/utils.ts:11]: What does this parameter do? It looks as if it is a flag to force the link closed or that the user set the property to close the link. If so, please adjust the naming of this parameter to be more verbose (e.g., `userClose` or `forceClose`.

**Review by jrthompson33 — COMMENTED** [review]
PR is almost ready - see minor comments on one edge case for multiple plot segments.

File: src/core/prototypes/links/utils.ts (hunk 2 above)
Comment (MrRamka) [reply src/core/prototypes/links/utils.ts]: Fixed, thanks!

File: src/core/prototypes/links/utils.ts (hunk 1 above)
Comment (MrRamka) [reply src/core/prototypes/links/utils.ts:26]: In this case, no. `linkThrough.plotSegment` has an `_id` on `plotSegment`. And in chart we have only one `plotSegment` with this `_id `

**Review by jrthompson33 — APPROVED** [review]
Approving PR, thanks for clarification on the unique id for PlotSegment.

**Pull request merged**